persistence/src/test/java/com/epam/rft/atsy/persistence/repositories/CandidateRepositoryIT.java

```diff
@@ -0,0 +1,323 @@
+        for (CandidateEntity candidate : POSITIVE_CANDIDATES_TEST) {
+            //Given
+            final String name = candidate.getName();
+            final String email = candidate.getEmail();
+            final String phone = candidate.getPhone();
+
+            //When
+            List<CandidateEntity> result = this.candidateRepository.findAllCandidatesByFilterRequest(
+                    name,
+                    email,
+                    phone,
+                    new Sort(createSortOrder(Sort.Direction.ASC, "phone")));
+
+            //Then
+            assertCandidateEntityListWithSorting(result, name, email, phone, createSortOrder(Sort.Direction.ASC, "phone"));
+        }
+    }
+
+    @Test
+    public void findAllCandidatesByFilterRequestWithValidDataSortedByPhoneInDescendingOrder() {
+        for (CandidateEntity candidate : POSITIVE_CANDIDATES_TEST) {
+            //Given
+            final String name = candidate.getName();
+            final String email = candidate.getEmail();
+            final String phone = candidate.getPhone();
+
+            //When
+            List<CandidateEntity> result = this.candidateRepository.findAllCandidatesByFilterRequest(
+                    name,
+                    email,
+                    phone,
+                    new Sort(createSortOrder(Sort.Direction.DESC, "phone")));
+
+            //Then
+            assertCandidateEntityListWithSorting(result, name, email, phone, createSortOrder(Sort.Direction.DESC, "phone"));
+        }
+    }
+
+    @Test
+    public void findAllCandidateByFilterRequestWithInvalidData() {
+        for (CandidateEntity candidate : NEGATIVE_CANDIDATES_TEST) {
+            //Given
+            final String name = candidate.getName();
+            final String email = candidate.getEmail();
+            final String phone = candidate.getPhone();
+
+            //When
+            List<CandidateEntity> result = this.candidateRepository.findAllCandidatesByFilterRequest(
+                    name,
+                    email,
+                    phone,
+                    null);
+
+            //Then
+            assertCandidateEntityListWithInvalidData(result);
+
+        }
+    }
+
+    private void assertCandidateEntityListWithInvalidData(List<CandidateEntity> candidates) {
+        assertThat(candidates, notNullValue());
+        assertThat(candidates, empty());
+    }
+
+    private void assertCandidateEntityListWithSorting(List<CandidateEntity> candidates,
+                                                      String expectedName,
+                                                      String expectedEmail,
+                                                      String expectedPhone,
+                                                      Sort.Order sorting) {
+
+        assertThat(candidates, notNullValue());
+
+        for (CandidateEntity entity : candidates) {
```

Comment:
Extract this for loop into a new method. Name the method something which represents the logic inside, eg: assertCandidateValues